kolibri/core/notifications/test/test_api.py

```diff
@@ -756,371 +876,261 @@ def test_parse_attemptslog_update_attempt_with_three_wrong_attempts(
             lesson_id=self.lesson_id,
             contentnode_id=self.node_1.id,
             reason=HelpReason.Multiple,
-            timestamp=attemptlog3.start_timestamp,
+            timestamp=attemptlog3.end_timestamp,
         )
 
     @patch("kolibri.core.notifications.api.create_notification")
     @patch("kolibri.core.notifications.api.save_notifications")
-    def test_update_lesson_assessment_with_three_wrong_attempts_on_same_attempt(
```

Comment:
Why the change from three to four? Three is the current expected behaviour.

Reply:
Oh, its because the help needed notification jumps when there are _more_ than 3 failed interaction (it was like that even before the refactor https://github.com/learningequality/kolibri/blob/c445798fad324c81fa076c78e1fb95b20ad6cb91/kolibri/core/notifications/api.py#L695). For now having `[{"type": "answer", "correct": 0} for _ in range(3)]` wasnt a problem, because we were creating more attemptlogs in the setup method, and we ended with four or more failed interactions in total.

So this test name was misleading because of that, but now I wonder what the spec is, and if the help needed really should jump with 3 or more failed interactions and what is wrong is the production code :thinking: (Even though the code have been like that for more than 6 years).

Reply:
Hrm, an excellent question. I was definitely always under the impression that it was for 3 or more, not more than three, I will do some digging and see if there is anything to back that up!